### DISCUSSION TITLE
How to put ads

### DISCUSSION THREAD

**Discussion by moisesnunes** [thuliteio/doks#935] [Q&A]
I am having trouble putting ads on the website. 
I created a folder google inside _layouts/partials/_ and then add two files **ads.html** e **analytics.html**, end put the google codes inside it.
But after that i am not sure where to put the 
``` 
{{ partial "google/ads" . }}
````
and 
```
{{ partial "google/analytics" . }}
```
I mean, which **single.html** should I put it in or what other file can I put it in so it will be rendered on all pages.

The link for the website: [https://www.aprendacpp.com/](url) and the source code  [https://github.com/moisesnunes/tutoriais](url)

thanks

**Comment by h-enk** [comment]
You could create a partial containing the ad (or analytics) code (e.g. like I did for getdoks.org: https://github.com/h-enk/getdoks.org/blob/master/layouts/partials/main/carbonads.html).

And call that partial when and where you want (e.g. like I did for getdoks.org: https://github.com/h-enk/getdoks.org/blob/master/layouts/docs/single.html#L20-L22). Note that, this could be in e.g. `./layouts/docs/single.html`, higher up in 
`./layouts/_default/single.html`, or for all pages (think ananlytics) in  `./layouts/_default/baseof.html`

Also, make sure to update your Content Security Policy (e.g. like I did for getdoks.org: https://github.com/h-enk/getdoks.org/blob/master/layouts/index.headers#L5)

Hope this helps

**Comment by macdolandson234** [comment]
Hi,
I am also looking for how can i add ads manually on a [site](https://thecourtdirect.com/old-bridge-municipal-court/). If you get answer mention me.
Thanks.
